### ISSUE TITLE
Noritz Floor Heater Switch Appears Stuck

### ISSUE THREAD

**Issue by aftertommy** [scottyphillips/echonetlite_homeassistant#110]
Our house has a Noritz bathtub and floor heating system. It has a single IP address, and appears to expose control over our bathtub heater and floor heating in 3 rooms.

When I try to turn on the floor heating for one of these rooms via the Home Assistant web ui, the device successfully turns on, but the web ui switch turns back to the OFF position. I am unable to toggle the switch again until I press the button on the physical controller.

Here is the logging I receive when I turn on the device from the web ui:

```
2022-09-01 12:18:33 DEBUG (MainThread) [custom_components.echonetlite] ECHONETLite Message Received - Processed data is {'EHD1': 16, 'EHD2': 129, 'TID': 11, 'SEOJGC': 2, 'SEOJCC': 123, 'SEOJCI': 3, 'DEOJGC': 5, 'DEOJCC': 255, 'DEOJCI': 1, 'ESV': 113, 'OPC': [{'EPC': 128, 'PDC': 0, 'EDT': b''}]}
```

Here are all of the devices that are exposed after adding the heater's IP address in Home Assistant:
<img width="1125" alt="image" src="https://user-images.githubusercontent.com/5618675/187825200-08fad49c-cae4-4a79-87f1-4ce4a9d35cc7.png">

This is the list of entities that are associated with the devices:

```
 sensor.water_heater_bath_heating_detection_status
 sensor.water_heater_bath_hot_water_adding_operation_setting
 sensor.water_heater_bath_hot_water_volume_setting_2
 sensor.water_heater_bath_hot_water_volume_setting_4
 sensor.water_heater_bath_hot_water_volume_setting_4_maximum_settable_level
 sensor.water_heater_bath_operation_status_monitor
 sensor.water_heater_bath_water_heater_status
 sensor.water_heater_bath_water_temperature_lowering_operation_setting
 sensor.water_heater_cumulative_amount_of_gas_consumption_measurement_value
 sensor.water_heater_detection_threshold_level
 sensor.water_heater_hot_water_heating_status
 sensor.water_heater_human_detection_status
 sensor.water_heater_measured_cumulative_amount_of_flowing_water
 sensor.water_heater_measured_temperature_value
 sensor.water_heater_operation_status
 sensor.water_heater_operation_status_2
 sensor.water_heater_operation_status_3
 sensor.water_heater_operation_status_4
 sensor.water_heater_operation_status_5
 sensor.water_heater_set_value_of_bath_temperature
 sensor.water_heater_set_value_of_hot_water_temperature
 sensor.water_heater_temperature_setting_2
 sensor.water_heater_temperature_setting_2_2
 sensor.water_heater_temperature_setting_2_3
 sensor.water_heater_unit_for_measured_cumulative_amounts_of_flowing_water
 switch.water_heater_bath_additional_boil_up_operation_setting
 switch.water_heater_bath_auto_mode_setting
 switch.water_heater_operation_status
 switch.water_heater_operation_status_2
 switch.water_heater_operation_status_3
 switch.water_heater_operation_status_4
```

**Comment by scottyphillips** [comment]
There is some behaviour issues related to the on/off switches that I think may have been resolved by @nao-pon as part of the 3.6.0 release. I don’t know if it is correctly setting the switch to on if it is configured for push notifications. You can try to force polling in the integrations options and see if that helps?

**Comment by aftertommy** [comment]
I performed a full upgrade to 3.6.0 (remove the component, reinstall ensuring it's 3.6.0, add the heater), and unfortunately no dice. Forcing polling doesn't appear to have resolved it.

**Comment by nao-pon** [comment]
I may have found a bug. the part shown below

https://github.com/scottyphillips/pychonet/blob/02cb1f4cdef6ed8adea5d120a74d66b1ea463cc0/pychonet/echonetapiclient.py#L35

probably correct

```python
esv_ok = esv in [0x71, 0x72, 0x73, 0x74]
```

**Comment by scottyphillips** [comment]
I’ve given you the god power @nao-pon san.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by aftertommy** [comment]
This appears to be working! Thank you!

**Comment by nao-pon** [comment]
This issue is resolved in pychonet 2.3.3, as per https://github.com/scottyphillips/echonetlite_homeassistant/issues/109#issuecomment-1236019741.

I will close this as the next version 3.6.1 will be released with this issue fixed. Thanks! 👍
